packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -212,27 +213,36 @@ export const createTransactionsTracker = (
       map((transactions) => transactions.filter(({ submittedAt }) => !!submittedAt)),
       mergeMap((transactions) => from(transactions))
     )
-  ).pipe(share());
+  ).pipe(
+    // Tx could be re-submitted, so we group by tx id
+    groupBy(({ tx: { id } }) => id),
```

Comment:
I don't understand. Could there be multiple tx with same id, so you can group them?

Reply:
Yes, this `groupBy` is needed to avoid "double tracking" the same transaction if it is re-submitted.
1. Submit tx (submitted$)
2. Re-submit tx (submitted$ same tx id) - without this change there would be 2 inner observables alive, so you would have 2 emissions on confirmed$, failed$ and inFlight$ as transaction state changes.